Xamarin.Forms.Platform.Android/Forms.cs

```diff
@@ -191,54 +193,149 @@ static Color GetAccentColor()
+				SetScreenOrientation();
+				_formsActivity.ConfigurationChanged += OnConfigurationChanged;
 			}
 
-			public override Size PixelScreenSize
+			public override Size PixelScreenSize => _pixelScreenSize;
+
+			public override Size ScaledScreenSize { get; }
+
+			public override double ScalingFactor => _scalingFactor;
+
+			public override void BeginDeviceOrientationNotifications()
 			{
-				get { return _pixelScreenSize; }
+				if (_androidOrientationEventListener != null)
+					return;
+
+				_androidOrientationEventListener = new AndroidOrientationEventListener((Context)_formsActivity, SensorDelay.Normal);
+				_androidOrientationEventListener.OrientationChanged += OnDeviceOrientationChanged;
+
+				if (_androidOrientationEventListener.CanDetectOrientation())
+					_androidOrientationEventListener.Enable();
 			}
 
-			public override Size ScaledScreenSize { get; }
+			public override void EndDeviceOrientationNotifications()
+			{
+				if (_androidOrientationEventListener == null)
+				    return;
+
+				_androidOrientationEventListener.OrientationChanged -= OnDeviceOrientationChanged;
+				_androidOrientationEventListener.Disable();
+				_androidOrientationEventListener.Dispose();
+				_androidOrientationEventListener = null;
+			}
+
+			void OnDeviceOrientationChanged(object sender, int i)
+			{
+				SetDeviceOrientation(i);
+			}
+
+			void OnConfigurationChanged(object sender, EventArgs e)
+			{
+				SetScreenOrientation();
+			}
 
-			public override double ScalingFactor
+			void SetDeviceOrientation(int rotation)
 			{
-				get { return _scalingFactor; }
+				if (rotation == OrientationEventListener.OrientationUnknown)
+				{
+					DeviceOrientation = DeviceOrientation.Unknown;
+					return;
+				}
+
+				if (PixelScreenSize.Width < PixelScreenSize.Height)
+				{
+					const int threshold = 45;
+
+					if (rotation <= threshold || rotation > 360 - threshold)
+						DeviceOrientation = DeviceOrientation.Portrait;
+					else if (rotation <= 360 - threshold && rotation > 270 - threshold)
+						DeviceOrientation = DeviceOrientation.Landscape;
+					else if (rotation <= 270 - threshold && rotation > 180 - threshold)
+						DeviceOrientation = DeviceOrientation.PortraitFlipped;
+					else if (rotation <= 180 - threshold && rotation > threshold)
+						DeviceOrientation = DeviceOrientation.LandscapeFlipped;
+				}
+				else if (PixelScreenSize.Width > PixelScreenSize.Height)
+				{
+					const int threshold = 45;
+
+					if (rotation <= threshold || rotation > 360 - threshold)
+						DeviceOrientation = DeviceOrientation.Landscape;
+					else if (rotation <= 360 - threshold && rotation > 270 - threshold)
+						DeviceOrientation = DeviceOrientation.PortraitFlipped;
+					else if (rotation <= 270 - threshold && rotation > 180 - threshold)
+						DeviceOrientation = DeviceOrientation.LandscapeFlipped;
+					else if (rotation <= 180 - threshold && rotation > threshold)
+						DeviceOrientation = DeviceOrientation.Portrait;
+				}
```

Comment:
I believe device orientation should work fine since it relies on sensors. Android seems to be ignoring screen orientation changes in multi window mode: https://developer.android.com/guide/topics/ui/multi-window.html#running

I think this sort of makes sense since it is possible that the width and height of the UI can be the same (on _device_ portrait or landscape) in which case you can't tell if it's portrait or landscape. Perhaps there are other reasons I don't know.

I should probably add `Unknown` to `ScreenOrientation` like its device counterpart so that if a form factor is not supported, it defaults to the unknown state.